config/main.py

```diff
@@ -3746,6 +3777,68 @@ def remove_profile(db, profile):
     else:
         ctx.fail("Profile {} doesn't exist".format(profile))
 
+@buffer.group(cls=clicommon.AbbreviationGroup)
+@click.pass_context
+def shared_headroom_pool(ctx):
+    """Configure buffer shared headroom pool"""
+    pass
+
+
+@shared_headroom_pool.command()
+@click.argument('ratio', metavar='<ratio>', type=int, required=True)
+@clicommon.pass_db
+def over_subscribe_ratio(db, ratio):
+    """Configure over subscribe ratio"""
+    config_db = db.cfgdb
+    ctx = click.get_current_context()
+
+    port_number = len(config_db.get_table('PORT'))
+    if ratio < 0 or ratio > port_number:
+        ctx.fail("Invalid over-subscribe-ratio value {}. It should be in range [0, {}]".format(ratio, port_number))
+
+    default_lossless_param = config_db.get_table("DEFAULT_LOSSLESS_BUFFER_PARAMETER")
+    first_item = True
+    for k, v in default_lossless_param.items():
+        if not first_item:
+            ctx.fail("More than one item in DEFAULT_LOSSLESS_BUFFER_PARAMETER table. Only the first one is updated")
+        first_item = False
+
+        if ratio == "0":
+            if "over_subscribe_ratio" in v.keys():
+                v.pop("over_subscribe_ratio")
+        else:
+            v["over_subscribe_ratio"] = ratio
```

Comment:
do we need to convert 'ratio' to a string?

Reply:
This will be done by `set_entry`. But I will check whether it's a common practice that to convert any type to int before pushing them into db. If so, I'll do it.

Reply:
Both behaviors are common so let's keep as it is.